jvm-streaming/src/Language/Java/Streaming.hs

```diff
@@ -48,37 +48,37 @@ isPoppableStream ref = do
         writeIORef ref (Streaming.cons x stream)
         return $ fromIntegral $ fromEnum True
 
-popStream :: Reflect a ty => IORef (Stream (Of a) IO ()) -> IO (J ty)
+popStream :: Reflect a ty => IORef (Stream (Of a) IO ()) -> IO (Ptr (J ty))
 popStream ref = do
     stream <- readIORef ref
     Streaming.uncons stream >>= \case
       Nothing -> do
         exc :: J ('Class "java.util.NoSuchElementException") <- new []
         JNI.throw exc
-        return jnull
+        return nullPtr
       Just (x, stream') -> do
         writeIORef ref stream'
-        reflect x
+        reflect x >>= JNI.getLocalRefPtr
```

Comment:
We need to return here a reference to a java object, but we want to make sure the reference is not deleted before java has a chance to use it. Therefore we call `getLocalRefPtr` here.

Perhaps you can find of other ways to arrange this.